src/llamafactory/train/ppo/workflow.py

```diff
@@ -54,7 +54,8 @@ def run_ppo(
     reward_model = create_reward_model(model, model_args, finetuning_args)
 
     # Initialize our Trainer
-    ppo_trainer = CustomPPOTrainer(
+    Trainer = factory_glm4v_trainer(CustomPPOTrainer) if model_args.visual_inputs_type == "glm4v_like" else CustomPPOTrainer
```

Comment:
If only the compute_loss function needs to be changed, I think you can use monkey_patch, such implementation is not very pretty

All such code has the same problem

Reply:
like `CustomPPOTrainer.compute_loss  = method_decorator(CustomPPOTrainer.compute_loss )`?

Reply:
yeah, you can test code like this